flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/BoundedBlockingSubpartitionWriteReadTest.java

```diff
@@ -234,15 +255,21 @@ private BoundedBlockingSubpartition createSubpartition() throws IOException {
 
 		private final int numBuffers;
 
-		LongReader(ResultSubpartitionView reader, long numLongs, int numBuffers) {
+		private final boolean compressEnabled;
+
+		private final BufferDecompressor decompressor;
```

Comment:
can we reuse the above class level `decompressor`?

Reply:
BufferDecompressor is not thread safe